botocore/handlers.py

```diff
@@ -880,6 +880,8 @@ def remove_subscribe_to_shard(class_attributes, **kwargs):
 
     ('before-parameter-build.s3.ListObjects',
      set_list_objects_encoding_type_url),
+    ('before-parameter-build.s3.ListObjectsV2',
+     set_list_objects_encoding_type_url),
```

Comment:
You also need to register the decoding after the response is received (line 945).

One handler injects the encoding type (`set_list_objects_encoding_type_url`), and the other handler (`decode_list_object`) decodes objects if we injected the encoding type param.  We need both.

Reply:
Ah got it.